### ISSUE TITLE
Failed to query available provider packages

### ISSUE THREAD

**Issue by aegladfelter** [terraform-google-modules/terraform-google-project-factory#679]
### TL;DR

When trying to install latest version of this module with terraform version `3.5.0` I get an error "Failed to query available provider packages" 

![Screen Shot 2022-01-26 at 9 40 42 PM](https://user-images.githubusercontent.com/23265287/151287944-74ba8f20-9329-4cd7-b8b6-0d760465c65b.png)

### Expected behavior

According to documentation, if I have version `3.5.0` of Terraform google provider installed on my machine running Linux Ubuntu, I should be able to use the `google-project-factory` module under version `10.1` as suggested in this repositories readme.

### Observed behavior

```
eglad22@terraform-testing-01:~$ sudo terraform init -upgrade
Upgrading modules...
Downloading registry.terraform.io/terraform-google-modules/project-factory/google 10.3.2 for project-factory...
- project-factory in .terraform/modules/project-factory
- project-factory.budget in .terraform/modules/project-factory/modules/budget
- project-factory.gsuite_group in .terraform/modules/project-factory/modules/gsuite_group
- project-factory.project-factory in .terraform/modules/project-factory/modules/core_project_factory
- project-factory.project-factory.project_services in .terraform/modules/project-factory/modules/project_services
- project-factory.quotas in .terraform/modules/project-factory/modules/quota_manager
- project-factory.shared_vpc_access in .terraform/modules/project-factory/modules/shared_vpc_access
Initializing the backend...
Initializing provider plugins...
- Finding hashicorp/google versions matching "3.5.0, >= 3.43.0, >= 3.50.0, < 4.0.0"...
- Finding hashicorp/google-beta versions matching ">= 3.1.0, >= 3.43.0, >= 3.50.0, < 4.0.0"...
- Finding hashicorp/null versions matching "~> 2.1"...
- Finding hashicorp/random versions matching "~> 2.2"...
- Installing hashicorp/google-beta v3.90.1...
- Installed hashicorp/google-beta v3.90.1 (signed by HashiCorp)
- Installing hashicorp/null v2.1.2...
- Installed hashicorp/null v2.1.2 (signed by HashiCorp)
- Installing hashicorp/random v2.3.1...
- Installed hashicorp/random v2.3.1 (signed by HashiCorp)
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider hashicorp/google: no available releases match the given constraints 3.5.0,
│ >= 3.43.0, >= 3.50.0, < 4.0.0
╵
```

### Terraform Configuration

```hcl
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "3.5.0"
    }
  }
}

module "project-factory" {
  source  = "terraform-google-modules/project-factory/google"
  version = "~> 10.1"

  name                 = "pf-test-1"
  random_project_id    = true
  org_id               = "1234567890"
  usage_bucket_name    = "pf-test-1-usage-report-bucket"
  usage_bucket_prefix  = "pf/test/1/integration"
  billing_account      = "ABCDEF-ABCDEF-ABCDEF"
  svpc_host_project_id = "shared_vpc_host_name"

  shared_vpc_subnets = [
    "projects/base-project-196723/regions/us-east1/subnetworks/default",
    "projects/base-project-196723/regions/us-central1/subnetworks/default",
    "projects/base-project-196723/regions/us-central1/subnetworks/subnet-1",
  ]
}

provider "google" {
  credentials = file("aeglad22-terraform-4d6bde06f4bc.json")
  project = "aeglad22-terraform"
  region  = "us-central1"
  zone    = "us-central1-c"
}
resource "google_compute_network" "vpc_network" {
  name = "terraform-network"
}
resource "google_compute_instance" "vm_instance" {
  name         = "terraform-instance"
  machine_type = "f1-micro"
  tags         = ["web", "dev"]
  boot_disk {
    initialize_params {
      image = "debian-cloud/debian-9"
    }
  }
  network_interface {
    network = google_compute_network.vpc_network.name
    access_config {
    }
  }
}
```
```


### Terraform Version

```sh
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v3.5.0
```

**Comment by aegladfelter** [comment]
I was mistaken trying to use google provider and this module in the same .tf file.
